src/PowerShellEditorServices/Services/Symbols/SymbolsService.cs

```diff
@@ -106,288 +107,178 @@ internal class SymbolsService
 
-                    foreach (IScriptExtent extent in references.OrderBy(e => e.StartOffset))
+                    _ = file.References.TryGetReferences(targetIdentifier, out ConcurrentBag<SymbolReference>? references);
+                    if (references is null)
                     {
-                        SymbolReference reference = new(
-                            SymbolType.Function,
-                            foundSymbol.SymbolName,
-                            extent);
-
-                        try
-                        {
-                            reference.SourceLine = file.GetLine(reference.ScriptRegion.StartLineNumber);
-                        }
-                        catch (ArgumentOutOfRangeException e)
-                        {
-                            reference.SourceLine = string.Empty;
-                            _logger.LogException("Found reference is out of range in script file", e);
-                        }
-                        reference.FilePath = file.FilePath;
-                        symbolReferences.Add(reference);
+                        continue;
                     }
 
-                    await Task.Yield();
-                    cancellationToken.ThrowIfCancellationRequested();
+                    symbols.AddRange(references);
                 }
             }
 
-            return symbolReferences;
+            return symbols;
         }
 
         /// <summary>
-        /// Finds all the occurrences of a symbol in the script given a file location
+        /// Finds all the occurrences of a symbol in the script given a file location.
         /// </summary>
-        /// <param name="file">The details and contents of a open script file</param>
-        /// <param name="symbolLineNumber">The line number of the cursor for the given script</param>
-        /// <param name="symbolColumnNumber">The column number of the cursor for the given script</param>
-        /// <returns>FindOccurrencesResult</returns>
-        public static IReadOnlyList<SymbolReference> FindOccurrencesInFile(
-            ScriptFile file,
-            int symbolLineNumber,
-            int symbolColumnNumber)
+        public static IEnumerable<SymbolReference>? FindOccurrencesInFile(
+            ScriptFile scriptFile, int line, int column)
         {
-            SymbolReference foundSymbol = AstOperations.FindSymbolAtPosition(
-                file.ScriptAst,
-                symbolLineNumber,
-                symbolColumnNumber);
+            SymbolReference? symbol = FindSymbolAtLocation(scriptFile, line, column);
 
-            if (foundSymbol == null)
+            if (symbol is null)
             {
                 return null;
             }
 
-            return AstOperations.FindReferencesOfSymbol(file.ScriptAst, foundSymbol).ToArray();
+            scriptFile.References.TryGetReferences(symbol.SymbolName, out ConcurrentBag<SymbolReference>? references);
+            return references;
```

Comment:
```suggestion
            return references.Where((r) =>
                    symbol.SymbolType == r.SymbolType ||
                    (symbol.SymbolType is SymbolType.Class or SymbolType.Enum or SymbolType.Type
                        && r.SymbolType is SymbolType.Class or SymbolType.Enum or SymbolType.Type) ||
                    (symbol.SymbolType is SymbolType.EnumMember or SymbolType.Property
                        && r.SymbolType is SymbolType.EnumMember or SymbolType.Property)
                );
```

Fix symbol highlight etc.